src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
-    { dimension: 'cube-array', arrayLayerCount: 12, _success: true }, // it is OK to create a cube map array texture view with arrayLayerCount % 6 == 0
-    // it is an error to create a cube map array texture view with arrayLayerCount % 6 != 0
-    { dimension: 'cube-array', arrayLayerCount: 11, _success: false },
-    { dimension: 'cube-array', arrayLayerCount: 13, _success: false },
-  ] as const)
-  .fn(async t => {
-    const { dimension = '2d-array', arrayLayerCount, _success } = t.params;
+g.test('mip_levels')
+  .desc(
+    `Views must have at least one level, and must be within the level of the base texture.
 
-    const texture = t.createTexture({ arrayLayerCount: 16 });
+  - mipLevelCount=0 at various baseMipLevel values
+  - Cases where baseMipLevel+mipLevelCount goes past the end of the texture
+  - Cases with baseMipLevel or mipLevelCount undefined (compares against reference defaulting impl)
+  `
+  )
+  .params(u =>
+    u
+      .combine('dimension', kTextureDimensions)
+      .beginSubcases()
+      .combineWithParams([
+        { textureLayers: 1, textureLevels: 1 },
+        { textureLayers: kLayers, textureLevels: kLevels },
+      ])
+      .combine('view', [
+        {},
+        { baseMipLevel: 0 },
+        { mipLevelCount: 1 },
+        { baseMipLevel: 0, mipLevelCount: 1 },
+        { baseMipLevel: 1, mipLevelCount: 1 },
+        { baseMipLevel: kLevels - 2, mipLevelCount: 1 },
+        { baseMipLevel: kLevels - 1, mipLevelCount: 1 },
+        { baseMipLevel: 0, mipLevelCount: 2 },
+        { baseMipLevel: 1, mipLevelCount: 2 },
+        { baseMipLevel: kLevels - 2, mipLevelCount: 2 },
+        // mipLevelCount == undefined means to use all remaining levels
+        { mipLevelCount: undefined, baseMipLevel: 0 },
+        { mipLevelCount: undefined, baseMipLevel: 1 },
+        { mipLevelCount: undefined, baseMipLevel: kLevels - 1 },
+        { mipLevelCount: undefined, baseMipLevel: kLevels },
+        // mipLevelCount == 0 means zero mip levels, which is never valid
+        { mipLevelCount: 0, baseMipLevel: 0 },
+        { mipLevelCount: 0, baseMipLevel: 1 },
+        { mipLevelCount: 0, baseMipLevel: kLevels - 1 },
+        { mipLevelCount: 0, baseMipLevel: kLevels },
+        // mip level range out of bounds
+        { baseMipLevel: 0, mipLevelCount: kLevels + 1 },
+        { baseMipLevel: 1, mipLevelCount: kLevels },
+        { baseMipLevel: kLevels - 1, mipLevelCount: 2 },
+        { baseMipLevel: kLevels, mipLevelCount: 1 },
+      ])
+  )
+  .fn(t => {
+    const { dimension, textureLayers, textureLevels, view } = t.params;
 
-    const descriptor = t.getDescriptor({
+    const textureDescriptor: GPUTextureDescriptor = {
+      format: 'rgba8unorm',
       dimension,
-      arrayLayerCount,
-    });
+      size: [32, 32, textureLayers],
+      mipLevelCount: textureLevels,
+      usage: GPUTextureUsage.SAMPLED,
+    };
+    const reified = reifyTextureViewDescriptor(textureDescriptor, view);
```

Comment:
Same here. I feel like that you want to test `2d, 2d-array, cube, and cube-array` view types, but it seems to me that this function always sets viewDimension to `2d` if texture is `2d`?